web/viewer.js

```diff
@@ -177,6 +177,11 @@ function getViewerConfiguration() {
 
 function webViewerLoad() {
   let config = getViewerConfiguration();
+  let locationHref = window.location.href;
+  let downloadButton = config.toolbar.download;
+  if (locationHref.startsWith('file://')) {
```

Comment:
This is checking the url, which is correct for my use-case, but has side effects when serving the `viewer.html` itself from `file`.

Reply:
OK,  I will look into this also once I'm back from college

Reply:
Furthermore you cannot use `String.prototype.startsWith()` without a polyfill since it's not properly supported across browsers, as shown in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith#Browser_compatibility.

Reply:
@Hirse , Could you please tell me all the possible cases of rendering `pdf.js`. Actually right now I'm trying to deeply understand the code. So if you tell me all the rendering cases and how can I try out those cases that would help me to understand the code better. By rendering cases what I mean is 

- opening viewer.html manually
- firefox extension. @Snuffleupagus , How do I test the firefox extension generated by me using `gulp firefox`
- what does the generic build do

Reply:
@himanish-star I don't know all possibilities, but you don't need all.
It is safe to assume that using `window.location` is not the best solution. You can look at the code where the PDF is loaded and check its URL there.

Reply:
@Hirse , thanks. I may take few more days but I'll surely try to fix this.